cmake/OpenCVFindIPP.cmake

```diff
@@ -34,6 +34,11 @@ unset(IPP_VERSION_MAJOR)
 unset(IPP_VERSION_MINOR)
 unset(IPP_VERSION_BUILD)
 
+if (X86 AND UNIX AND NOT APPLE AND NOT ANDROID AND BUILD_SHARED_LIBS)
+    message(STATUS "On 32-bit Linux IPP can not currently be used with dynamic libs. Set BUILD_SHARED_LIBS=OFF")
```

Comment:
@mshabunin answer is pretty much complete. I'd just add that disabling one certain function, listed in the bug report, does not solve the problem. You immediately get another linker error and it goes on and on. I tried to disable at least 10 functions and then gave up.